src/main/java/com/autotune/analyzer/performanceProfiles/PerformanceProfileInterface/ResourceOptimizationOpenshiftImpl.java

```diff
@@ -83,36 +80,26 @@ public List<KruizeRecommendationEngine> getEngines() {
-            the CreateExperiment API must adhere strictly to the trail settings' measurement duration and should not allow arbitrary values
+             The general strategy involves initially attempting the optimal query;
         */
-        String experiment_name = kruizeObject.getExperimentName();
-        int theoriticalLimitRows = (int) ((
-                KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS *
-                        KruizeConstants.DateFormats.MINUTES_FOR_DAY)
-                / kruizeObject.getTrial_settings().getMeasurement_durationMinutes_inDouble());
-
-        // Adding a 10% of value to make sure there are enough data points to gather the duration required
-        // Will be deprecated once we extract exact number of rows from DB based on duration in minutes needed for each term
-        int practicalLimitRows = RecommendationUtils.getThreshold(theoriticalLimitRows, 10, true);
-
-        if (null != interval_start_time) {
-            long diffMilliseconds = interval_end_time.getTime() - interval_start_time.getTime();
-            long minutes = diffMilliseconds / (60 * 1000);
-            int addToLimitRows = (int) (minutes / kruizeObject.getTrial_settings().getMeasurement_durationMinutes_inDouble());
-            LOGGER.debug("add to limit rows set to {}", addToLimitRows);
-            practicalLimitRows = practicalLimitRows + addToLimitRows;
-        }
-        LOGGER.debug("Limit rows set to {}", practicalLimitRows);
-
-
+        // Convert the Timestamp to a Calendar instance in UTC time zone
+        Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        cal.setTimeInMillis(interval_end_time.getTime());
+        /*
+         * interval_start_time Subtract (LONG_TERM_DURATION_DAYS +  THRESHOLD days)
+         * Incorporate a buffer period of "threshold days" to account for potential remote cluster downtime.
+         * This adjustment aims to align the cumulative hours' duration with LONG_TERM_DURATION_DAYS.
+         */
+        cal.add(Calendar.DAY_OF_MONTH, -(KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS +
+                KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS_THRESHOLD));
+        // Get the new Timestamp after subtracting 10 days
+        interval_start_time = new Timestamp(cal.getTimeInMillis());
         Map<String, KruizeObject> mainKruizeExperimentMap = new HashMap<>();
+        String experiment_name = kruizeObject.getExperimentName();
         mainKruizeExperimentMap.put(experiment_name, kruizeObject);
         new ExperimentDBService().loadResultsFromDBByName(mainKruizeExperimentMap,
                 experiment_name,
-                interval_end_time,
-                practicalLimitRows);
+                interval_start_time,interval_end_time);
```

Comment:
interval_start_time already has a different meaning, so this needs to be called something else